client/api/disk/v1alpha1/api.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+service Disk {
+    // ListDiskLocations returns locations <Adapter, Bus, Target, LUN ID> of all
+    // disk devices enumerated by the host
+    rpc ListDiskLocations(ListDiskLocationsRequest) returns (ListDiskLocationsResponse) {}
+
+    // PartitionDisk initializes and partitions a disk device (if the disk has not
+    // been partitioned already) and returns the resulting volume device ID
+    rpc PartitionDisk(PartitionDiskRequest) returns (PartitionDiskResponse) {}
+
+    // Rescan refreshes the host's storage cache
+    rpc Rescan(RescanRequest) returns (RescanResponse) {}
+}
+
+message ListDiskLocationsRequest {
+    // Intentionally empty
+}
+
+message DiskLocation {
+    string Adapter = 1;
+    string Bus = 2;
+    string Target = 3;
+    string LUNID = 4;
+}
+
+message ListDiskLocationsResponse {
+    // Map of disk device IDs and <adapter, bus, target, lun ID> associated with each disk device
+    map <string, DiskLocation> disk_locations = 1;
+}
+
+message PartitionDiskRequest {
+    // Disk device ID of the disk to partition
+    string diskID = 1;
+}
+
+message PartitionDiskResponse {
+    // Intentionally empty
```

Comment:
we don't need any result from partition?

Reply:
For errors, the request can directly return it (e.g. at https://github.com/kubernetes-csi/csi-proxy/pull/15/files#diff-87dc7f1dc074717f7df0d9698f32ce1fR79).

To obtain the Windows volume objects (resulting from partition creation on a disk), a plugin is expected to use `ListVolumesOnDisk`: https://github.com/kubernetes-csi/csi-proxy/blob/master/client/api/volume/v1alpha1/api.proto#L8